src/d/actor/d_a_obj_eff.cpp

```diff
@@ -2,261 +2,479 @@
+        static JGeometry::TVec3<f32> em_scl(1.0f, 0.0f, 1.0f);
+        emitter->setEmitterScale(em_scl);
+        static JGeometry::TVec3<f32> em_trans(0.0f, 5.0f, 0.0f);
+        emitter->setEmitterTranslation(em_trans);
+        ret = true;
+    } else {
+        ret = false;
+    }
+
+    return ret;
 }
 
 /* 00000D24-00000D98       .text eff_set_pinecone_smoke__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set_pinecone_smoke() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set_pinecone_smoke() {
+    dPa_control_c *particle = g_dComIfG_gameInfo.play.getParticle();
+    JPABaseEmitter *emitter =
+            particle->set(dPa_control_c::dPtclGroup_Toon_e, dPa_name::ID_SCENE_A16B, &current.pos, NULL, NULL, 0xb4,
+                          mParticleCallback, -1, NULL, NULL, NULL);
+
+    return emitter != NULL;
 }
 
 /* 00000D98-00000E80       .text eff_set_woodBox_smoke__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set_woodBox_smoke() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set_woodBox_smoke() {
+    static cXyz particle_scl(2.5f, 2.5f, 1.0f);
+    dPa_control_c *particle = g_dComIfG_gameInfo.play.getParticle();
+    JPABaseEmitter *emitter = particle->set(dPa_control_c::dPtclGroup_Toon_e, dPa_name::ID_COMMON_2027, &current.pos,
+                                            NULL, NULL, 0xb4, mParticleCallback, -1, NULL, NULL, &particle_scl);
+
+    if (emitter != NULL) {
+        emitter->mRate = 30.0f;
+        emitter->mMaxFrame = 1;
+        return true;
+    }
+
+    return false;
 }
 
 /* 00000E80-00000F6C       .text eff_set__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set() {
+    static Proc proc[] = {
+        &eff_set_barrel_smoke,
+        &eff_set_stool_smoke,
+        &eff_set_skull_smoke,
+        &eff_set_land_smoke,
+        &eff_set_pinecone_smoke,
+        &eff_set_woodBox_smoke
+    };
+
+    return (this->*proc[mProcIndex])();
 }
 
 /* 00000F6C-00001030       .text _create__Q28daObjEff5Act_cFv */
 cPhs_State daObjEff::Act_c::_create() {
-    /* Nonmatching */
+    static const u32 heap_size[] = {
+        0x00000040,
+        0x00000040,
+        0x00000040,
+        0x00000020,
+        0x00000040,
+        0x00000040,
+    };
+
+    mProcIndex = daObj::PrmAbstract(this, PRM_ARG_W, PRM_ARG_S);
+    fopAcM_SetupActor(this, daObjEff::Act_c);
+    int phase = cPhs_COMPLEATE_e;
+    if (fopAcM_entrySolidHeap(this, &solidHeapCB, heap_size[mProcIndex])) {
+        if ((u8) eff_set() != 0) {
+            fopDwTg_DrawQTo(&draw_tag);
+        } else {
+            phase = cPhs_ERROR_e;
+        }
+    } else {
+        phase = cPhs_ERROR_e;
+    }
+
+    return phase;
 }
 
 /* 00001030-00001068       .text remove_barrel_smoke__Q28daObjEff5Act_cFv */
 void daObjEff::Act_c::remove_barrel_smoke() {
-    /* Nonmatching */
+    if (mParticleCallback != NULL) {
+        static_cast<BarrelSmokeCB *>(mParticleCallback)->end();
```

Comment:
These `end` calls use the `remove` inline according to the debug maps
```suggestion
        mParticleCallback->remove();
```